h2o-core/src/main/java/water/AutoBuffer.java

```diff
@@ -1359,7 +1359,7 @@ public AutoBuffer putA8( long[] ary ) {
 
     put1(8);                    // Ship as full longs
     int sofar = x;
-    if ((y-sofar)*8 > _bb.remaining()) expandByteBuffer(ary.length*8);
+    if ((y-sofar)*8 > _bb.remaining()) expandByteBuffer(nzlen*8);
```

Comment:
@michalkurka method `putA8` tries to optimize the communication and trims the input array by leading and trailing zeros. New array size is `nzlen` and we should expand the byte buffer according to that

Reply:
@jakubhava great investigation, thank you!